### ISSUE TITLE
Crash on ApptentiveCoversationRequest

### ISSUE THREAD

**Issue by gringoireDM** [apptentive/apptentive-ios#206]
Hello, Since the update to the latest SDK our users are experiencing crashes due to nil parameters inserted in dictionary in the method JSONDictionary for the class ApptentiveConversationRequest.

The steps to reproduce the issue unfortunately are unclear, but the number of crashes is actually growing.

Attached the crashlog:

```

Fatal Exception: NSInvalidArgumentException*** -[__NSPlaceholderDictionary initWithObjects:forKeys:count:]: attempt to insert nil object from objects[0] Raw Text
--
 
Fatal Exception: NSInvalidArgumentException
0  CoreFoundation                 0x18f1e91b8 __exceptionPreprocess
1  libobjc.A.dylib                0x18dc2055c objc_exception_throw
2  CoreFoundation                 0x18f0cceac -[__NSPlaceholderDictionary initWithObjects:forKeys:count:]
3  CoreFoundation                 0x18f0ccd1c +[NSDictionary dictionaryWithObjects:forKeys:count:]
4  Apptentive                     0x10088d610 -[ApptentiveConversationRequest JSONDictionary] (ApptentiveConversationRequest.m:46)
5  Apptentive                     0x1008c5c30 -[ApptentiveRequest payload] (ApptentiveRequest.m:39)
6  Apptentive                     0x100881140 -[ApptentiveClient URLRequestWithRequest:] (ApptentiveClient.m:97)
7  Apptentive                     0x100880e5c -[ApptentiveClient requestOperationWithRequest:token:delegate:] (ApptentiveClient.m:72)
8  Apptentive                     0x10088615c -[ApptentiveConversationManager fetchConversationToken:] (ApptentiveConversationManager.m:315)
9  Apptentive                     0x100884e08 -[ApptentiveConversationManager loadConversation] (ApptentiveConversationManager.m:199)
10 Apptentive                     0x100884c50 -[ApptentiveConversationManager loadActiveConversation] (ApptentiveConversationManager.m:98)
11 Apptentive                     0x10087d6d4 -[ApptentiveBackend startUp] (ApptentiveBackend.m:349)
12 Apptentive                     0x10087b974 __88-[ApptentiveBackend initWithApptentiveKey:signature:baseURL:storagePath:operationQueue:]_block_invoke.54 (ApptentiveBackend.m:130)
13 Foundation                     0x18fcd7fb0 __NSBLOCKOPERATION_IS_CALLING_OUT_TO_A_BLOCK__
14 Foundation                     0x18fc1caa8 -[NSBlockOperation main]
15 Foundation                     0x18fc0d0a4 -[__NSOperationInternal _start:]
16 Foundation                     0x18fcda35c __NSOQSchedule_f
17 libdispatch.dylib              0x18e0721bc _dispatch_client_callout
18 libdispatch.dylib              0x18e0803dc _dispatch_queue_serial_drain
19 libdispatch.dylib              0x18e0759a4 _dispatch_queue_invoke
20 libdispatch.dylib              0x18e08234c _dispatch_root_queue_drain
21 libdispatch.dylib              0x18e0820ac _dispatch_worker_thread3
22 libsystem_pthread.dylib        0x18e27b2a0 _pthread_wqthread
23 libsystem_pthread.dylib        0x18e27ad8c start_wqthread
```

**Comment by frankus** [comment]
Thanks for the report. We'll investigate and keep you updated. Can you let us know the exact version of the SDK that you're using?

**Comment by gringoireDM** [comment]
4.0.5

**Comment by gringoireDM** [comment]
Updated the SDK to 4.0.6. The crash still occurs.

**Comment by frankus** [comment]
I just released 4.0.7 which will should prevent this crash from occurring.

**Comment by frankus** [comment]
I'm closing this for the time being. Please reach out of you experience this issue with 4.0.7.
